docs(env-variables): update docs to use T3 Env #1384 (#1391)
t3-oss/create-t3-app: www/src/pages/pt/usage/env-variables.mdx

---
title: Variáveis de Ambiente
description: Iniciando com Create T3 App 
layout: ../../../layouts/docs.astro
lang: pt
isMdx: true
---

import Callout from "../../../components/docs/callout.tsx";

> O Create T3 App usa sua mais nova biblioteca [@t3-oss/env-nextjs](https://env.t3.gg) e [zod](https://zod.dev) para validar variáveis de ambiente durante o tempo de execução (runtime) _e_ buildtime fornecendo uma lógica simples no arquivo `src/env.ts`:

Reviewer: ```suggestion
O Create T3 App usa sua mais nova biblioteca [@t3-oss/env-nextjs](https://env.t3.gg) e [zod](https://zod.dev) para validar variáveis de ambiente durante o tempo de execução (runtime) _e_ buildtime fornecendo uma lógica simples no arquivo `src/env.mjs`:
```

> ## env.ts

Reviewer: ```suggestion
## env.mjs
```

> _Muito longo, não li; Se você quiser adicionar uma nova variável de ambiente, você deve adicioná-la ao seu `.env` ao definir no validador em `src/env.ts`._

Reviewer: ```suggestion
_Muito longo, não li; Se você quiser adicionar uma nova variável de ambiente, você deve adicioná-la ao seu `.env` ao definir no validador em `src/env.mjs`._
```

Este arquivo é dividido em duas partes - o esquema e a desestruturação do objeto, bem como a lógica de validação. A lógica de validação não deve ser tocada.

> ```ts:env.ts

Reviewer: ```suggestion
```ts:env.mjs
```

import { createEnv } from "@t3-oss/env-nextjs";
import { z } from "zod";
 
export const env = createEnv({
  server: {
    NODE_ENV: z.enum(["development", "test", "production"]),
  },
  client: {
    // NEXT_PUBLIC_CLIENTVAR: z.string(),
  },
  runtimeEnv: {
    NODE_ENV: process.env.NODE_ENV,
  },
});
```

T3 Env traz um novo conceito através do `createEnv`, o qual é responsável por criar o esquema e irá incluir a lógica de validação para  variáveis de ambiente no cliente e servidor. 

<Callout type="info">
    Para mais informações sobre como o `createEnv` funciona internamente, confira a documentação de [T3 Env](https://env.t3.gg/docs/introduction)
</Callout>

## Usando Variáveis de Ambiente

> Quando você quiser usar suas variáveis de ambiente, você pode importá-las de `env.ts` (ou `env.mjs`) e usá-las como faria normalmente. Se você importar isso no cliente e tentar acessar uma variável de ambiente do lado do servidor, receberá um erro em tempo de execução (runtime).

Reviewer: ```suggestion
Quando você quiser usar suas variáveis de ambiente, você pode importá-las de `env.mjs` e usá-las como faria normalmente. Se você importar isso no cliente e tentar acessar uma variável de ambiente do lado do servidor, receberá um erro em tempo de execução (runtime).
```

```ts:pages/api/hello.ts

> import { env } from "../../env.ts";

Reviewer: ```suggestion
import { env } from "../../env.mjs";
```

// `env` é totalmente tipado e fornece preenchimento automático 
const dbUrl = env.DATABASE_URL;
```

```ts:pages/index.tsx

> import { env } from "../env.ts";

Reviewer: ```suggestion
import { env } from "../env.mjs";
```

📄 `.env`: Insira sua variável de ambiente como faria normalmente em um arquivo `.env`, ou seja, `CHAVE=VALOR`

> 📄 `env.ts`: Adicione a lógica de validação apropriada para as variáveis de ambiente definindo um esquema Zod, por exemplo `TECLA: z.string()`

Reviewer: ```suggestion
📄 `env.mjs`: Adicione a lógica de validação apropriada para as variáveis de ambiente definindo um esquema Zod, por exemplo `NOME_DA_VARIAVEL: z.string()`
```

Opcionalmente, você também pode manter `.env.example` atualizado:

📄 `.env.example`: Insira sua variável de ambiente, mas certifique-se de não incluir o valor se for secreto, ou seja, `CHAVE=VALOR` ou `CHAVE=`

### Exemplo

_Quero adicionar meu token da API do Twitter como uma variável de ambiente do lado do servidor_

1. Adicione a variável de ambiente a `.env`:

```
TWITTER_API_TOKEN=1234567890
```

> 2. Adicione a variável de ambiente a `env.ts`:

Reviewer: ```suggestion
2. Adicione a variável de ambiente a `env.mjs`:
```

> ```ts

Reviewer: ```suggestion
```ts:env.mjs
```

import { createEnv } from "@t3-oss/env-nextjs";
import { z } from "zod";
 
export const env = createEnv({
  server: {

>     TWITTER_API_TOKEN: z.enum(["development", "test", "production"]),

Reviewer: ```suggestion
    TWITTER_API_TOKEN: z.string(),
    NODE_ENV: z.enum(["development", "test", "production"])
```
